src/main/java/org/opensearch/neuralsearch/ml/MLCommonsClientAccessor.java

```diff
@@ -69,79 +100,164 @@ public void inferenceSentence(
     }
 
     /**
-     * Abstraction to call predict function of api of MLClient with default targetResponse filters. It uses the
-     * custom model provided as modelId and run the {@link FunctionName#TEXT_EMBEDDING}. The return will be sent
-     * using the actionListener which will have a {@link List} of {@link List} of {@link Float} in the order of
-     * inputText. We are not making this function generic enough to take any function or TaskType as currently we
-     * need to run only TextEmbedding tasks only.
+     * Abstraction to call predict function of api of MLClient with default targetResponse filters. It
+     * uses the custom model provided as modelId and run the {@link FunctionName#TEXT_EMBEDDING}. The
+     * return will be sent using the actionListener which will have a {@link List} of {@link List} of
+     * {@link Float} in the order of inputText. We are not making this function generic enough to take
+     * any function or TaskType as currently we need to run only TextEmbedding tasks only.
      *
-     * @param modelId {@link String}
+     * @param modelId   {@link String}
      * @param inputText {@link List} of {@link String} on which inference needs to happen
-     * @param listener {@link ActionListener} which will be called when prediction is completed or errored out
+     * @param listener  {@link ActionListener} which will be called when prediction is completed or
+     *                  errored out
      */
     public void inferenceSentences(
         @NonNull final String modelId,
         @NonNull final List<String> inputText,
         @NonNull final ActionListener<List<List<Float>>> listener
     ) {
-        inferenceSentences(TARGET_RESPONSE_FILTERS, modelId, inputText, listener);
+        inferenceSentences(TARGET_RESPONSE_FILTERS, modelId, inputText, null, listener);
```

Comment:
Why null is passed here?

Reply:
Can you create a DTO object here. And add the values which are absolutely necessary to pass. That way you can avoid sending null here.

Reply:
> Can you create a DTO object here. And add the values which are absolutely necessary to pass. That way you can avoid sending null here.

Let's do it in the next PR.

Reply:
@vibrantvarun This is been tracked in this issue: https://github.com/opensearch-project/neural-search/issues/790